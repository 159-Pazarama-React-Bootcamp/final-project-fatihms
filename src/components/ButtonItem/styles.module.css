.button-item {
  background-color: #112031;
  border: none;
  border-radius: 4px;
  color: #fff;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  margin-top: 10px;
  width: 100%;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.button-item:hover {
  background-color: #2f4247;
}
