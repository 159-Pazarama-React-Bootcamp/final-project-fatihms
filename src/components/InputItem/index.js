import React from "react";

import PropTypes from "prop-types";

import styles from "./styles.module.css";
/*eslint-disable */

function InputItem(props) {
  const {
    id,
    name,
    type,
    placeholder,
    onChange,
    value,
    labelText = "",
    error,
    touched,
  } = props;
  return (
    <div className={styles["input-container"]}>
      <input
        id={id}
        name={name}
        type={type}
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        style={error && touched ? { border: "2px solid red" } : null}
      />
      <label htmlFor={id}>{`${labelText} ${
        error && touched && error ? "- " + error : ""
      }`}</label>
    </div>
  );
}

export default InputItem;

InputItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  labelText: PropTypes.string.isRequired,
  error: PropTypes.string.isRequired,
  touched: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
};

InputItem.defaultProps = {
  error: "",
  touched: "",
};
