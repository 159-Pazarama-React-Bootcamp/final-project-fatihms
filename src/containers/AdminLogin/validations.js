import * as yup from "yup";

const validations = yup.object().shape({
  username: yup.string().required("!kullanıcı adı gerekli"),
  password: yup.string().required("!şifre gerekli"),
});

export default validations;
