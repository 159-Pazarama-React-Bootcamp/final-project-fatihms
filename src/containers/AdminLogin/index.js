import React from "react";

import FormContainer from "../../components/FormContainer";
import InputItem from "../../components/InputItem";

import styles from "./styles.module.css";

function AdminLogin() {
  return (
    <section className={styles["admin-page"]}>
      <FormContainer width="30">
        <InputItem
          id="email"
          name="email"
          type="email"
          placeholder="Email"
          labelText="Email"
        />
      </FormContainer>
    </section>
  );
}

export default AdminLogin;
