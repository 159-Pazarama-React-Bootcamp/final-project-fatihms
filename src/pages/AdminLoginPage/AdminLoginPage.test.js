/*eslint-disable */

import React from "react";

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

import { mount, shallow } from "enzyme";

import AdminLoginPage from "./index";
import AdminLogin from "../../containers/AdminLogin";
import ButtonItem from "../../components/ButtonItem";
import InputItem from "../../components/InputItem";

describe("AdminLoginPage", () => {
  beforeEach(() => {
    render(
      <Router>
        <AdminLoginPage />
      </Router>
    );
  });
  test("AdminLoginPage renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <AdminLoginPage />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("giriş yap button renders correctly", () => {
    expect(screen.getByText("Giriş Yap")).toBeInTheDocument();
  });
});
