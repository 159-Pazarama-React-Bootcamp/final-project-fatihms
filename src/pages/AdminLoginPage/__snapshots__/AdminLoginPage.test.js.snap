// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLoginPage AdminLoginPage renders correctly 1`] = `
<section
  className="admin-page"
>
  <div
    className="form-container form-container--20"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-container"
      >
        <input
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="admin"
          style={null}
          type="text"
          value=""
        />
        <label
          htmlFor="username"
        >
          Kullanıcı Adı 
        </label>
      </div>
      <div
        className="input-container"
      >
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="*********"
          style={null}
          type="password"
          value=""
        />
        <label
          htmlFor="password"
        >
          Şifre 
        </label>
      </div>
      <button
        className="button-item"
        type="submit"
      >
        Giriş Yap
      </button>
    </form>
  </div>
</section>
`;
